go/roothash/api/commitment/pool.go

```diff
@@ -465,6 +468,13 @@ func (p *Pool) ProcessCommitments(didTimeout bool) (OpenCommitment, error) {
 		}
 
 		// No majority commitment so far.
+	case false:
+		// In case the number of failures is bigger than the number of allowed stragglers, round
+		// will proceed to discrepancy resolution mode regardless of additional commits.
```

Comment:
```suggestion
		// If it is already known that the number of valid commitments will not exceed
		// the required threshold, there is no need to wait for the timer to expire.
		// Instead, proceed directly to the discrepancy resolution mode, regardless
		// of any additional commits.
```
[nit] I would rather comment why not what.